t/test-lib-event.sh

```diff
@@ -55,51 +56,56 @@ projfs_event_printf () {
 		err=$("$TEST_DIRECTORY"/get_strerror "$1"); shift
 	else
 		err=""
-		err_msg=""
 	fi
 
 	eval msg=\$event_msg_"$1"
 	eval code=\$event_"$2"
 
 	if test ":$4" = ":"
 	then
-		target=""
+		target_msg=""
 	else
-		target=", $4"
+		target_msg=", $4"
 	fi
-	log_msg=$(printf "$LOG_FMT" \
-		"$msg" "$3" "$target" "$code" "$EXEC_PID_MARK")
+	out_msg=$(printf "$OUT_FMT" \
+		"$msg" "$3" "$target_msg" "$code" "$EXEC_PID_MARK")
 
-	event_log_msgs="${event_log_msgs:+$event_log_msgs$NL}$log_msg"
+	event_out_msgs="${event_out_msgs:+$event_out_msgs$NL}$out_msg"
 
 	if test ":$err" != ":"
 	then
-		err_msg=$(printf "$ERR_FMT" \
+		if test ":$4" = ":"
```

Comment:
I wonder — is the `":x" = ":"` trick necessary, even in POSIX shell? Does this work? It appears to in local testing, as the double quotes are enough to ensure both take argument positions. (as opposed to `test $4 = …`, which will fail entirely if `$4` is unset.)

```suggestion
		if test "$4" = ""
```

Reply:
Good point -- We definitely remove this bit of [cargo cult](https://stackoverflow.com/a/6853353) shell programming, leftover from my [earliest days](https://terminals-wiki.org/wiki/index.php/DEC_LA120).

![DEC_LA120-2](https://user-images.githubusercontent.com/28857117/58737967-87f96380-83b8-11e9-9450-2aa9b66bd3f1.jpg)